tests/integration/external/guzzle6/test_timeout_async.php

```diff
@@ -20,6 +20,9 @@
 newrelic.code_level_metrics.enabled = 0
 */
 
+/*EXPECT_METRICS_EXIST
+External/127.0.0.1/all
```

Comment:
I'm unclear on why the metric and segment name below would contain "127.0.0.1" as the host but the `http.url` is captured with `ENV[EXTERNAL_HOST]` which I'm guessing is a different value as it necessitated adding support for pulling in env vars for `EXPECT_SPAN_EVENTS_LIKE`.

Reply:
`EXTERNAL_HOST` is set [here](https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/daemon/cmd/integration_runner/main.go#L324) and includes port number, which, unless provided via [`-external_port` flag](https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/daemon/cmd/integration_runner/main.go#L55-L57), is assigned at random (see [here](https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/daemon/cmd/integration_runner/main.go#L317-L319) and [doc](https://pkg.go.dev/net#Listen)). I was wondering if I should split off the port part into `EXTERNAL_HOST_PORT` in order to be able to use `ENV[EXTERNAL_HOST]` in `EXPECT_METRICS_EXIST`. However this would require me to change a lot of other tests which assume that `EXTERNAL_HOST` has the port number included. Thoughts?

Reply:
That is another consideration - I'm more wondering if `EXTERNAL_HOST` is actually "127.0.0.1" in all cases or not - and if the "127.0.0.1" in the metric line might not need to have an env substitution as well?  Also should the IP/host in the metric always be the same as in the `http.url`?

Reply:
> I'm more wondering if `EXTERNAL_HOST` is actually "127.0.0.1" in all cases or not - and if the "127.0.0.1" in the metric line might not need to have an env substitution as well?

It is not always 127.0.0.1: https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/tests/integration/external/guzzle7/test_transfer_exception_sync.php#L25

It is 127.0.0.1 for tests that make requests to endpoints exposed by `integration_runner`.

Reply:
Moreover, there are other tests that have 127.0.0.1 hardcoded in names but don't test "http.url" attributes. E.g.: https://github.com/newrelic/newrelic-php-agent/blob/55ab2076d11456cef2aed9d39f5bd87885679473/tests/integration/external/curl_exec/test_dt_correct_span_is_external.php#L69.

Reply:
Thank you for exploring this further.   I'm happy with your explanation.